src/core_atmosphere/dynamics/mpas_atm_time_integration.F

```diff
@@ -2759,45 +2815,71 @@ subroutine atm_recover_large_step_variables_work(nCells, nEdges, nCellsSolve, nE
       integer :: i, iCell, iEdge, k, cell1, cell2
       real (kind=RKIND) :: invNs, rcv, p0, flux
 
+      MPAS_ACC_TIMER_START('atm_recover_large_step_variables [ACC_data_xfer]')
+      !$acc enter data copyin(rho_p_save,rho_pp,rho_base,rw_save,rw_p, &
+      !$acc                   rtheta_p_save,rtheta_pp,rtheta_base, &
+      !$acc                   ru_save,ru_p) &
+      !$acc            create(rho_zz,rho_p,rw,w,rtheta_p,theta_m, &
+      !$acc                   ru,u) &
+      !$acc            copyin(wwAvg,ruAvg)
+      if (rk_step == 3) then
+         !$acc enter data copyin(rt_diabatic_tend,exner_base) &
+         !$acc            create(exner,pressure_p)
+      end if
+      MPAS_ACC_TIMER_STOP('atm_recover_large_step_variables [ACC_data_xfer]')
 
       rcv = rgas/(cp-rgas)
       p0 = 1.0e+05  ! this should come from somewhere else...
 
-      ! Avoid FP errors caused by a potential division by zero below by 
+      ! Avoid FP errors caused by a potential division by zero below by
       ! initializing the "garbage cell" of rho_zz to a non-zero value
+      !$acc parallel
```

Comment:
This might have been something that we talked about as a potential "best practice" after this PR was started, but would it be worth adding `default(present)` clauses to `parallel` directives?

Reply:
I'll get them added. I also think this PR pre-dates that discussion.

I think the `default(present)` directives are useful right now as we're porting and when we get to the point where we want to fuse data regions. Once the port is "finished" I think these clauses may cause friction whenever new code gets added. (Even that might be a benefit - e.g. if a physics change makes the `scalars` array partially present because it only modifies moisture species, that should be addressed before the PR is accepted.)

Reply:
Addressed in [cc7e7b7](https://github.com/MPAS-Dev/MPAS-Model/pull/1220/commits/cc7e7b7d9ec1887d3b9860ad71398f2dedadaa5b).